src/regtech_cleanup_api/routers/institution_cleanup.py

```diff
@@ -43,29 +43,33 @@ def delete_institution(request: Request, lei: str):
             detail=f"{lei} not valid test lei.",
         )
     else:
-        delete_domains = repo.delete_domains_by_lei(request.state.db_session, lei)
-        if not delete_domains:
-            logger.error(f"Domain(s) for LEI {lei} not deleted.")
+        return delete_helper(lei, request.state.db_session)
 
-        delete_sbl_type = repo.delete_sbl_type_by_lei(request.state.db_session, lei)
-        if not delete_sbl_type:
-            logger.error(f"sbl type(s) for LEI {lei} not deleted.")
 
-        res = repo.delete_institution(request.state.db_session, lei)
-        if not res:
+def delete_helper(lei: str, session: Session):
+    delete_domains = repo.delete_domains_by_lei(session, lei)
+    if not delete_domains:
+        logger.error(f"Domain(s) for LEI {lei} not deleted.")
+
+    delete_sbl_type = repo.delete_sbl_type_by_lei(session, lei)
+    if not delete_sbl_type:
+        logger.error(f"sbl type(s) for LEI {lei} not deleted.")
```

Comment:
Same.

Reply:
Honestly looking at the repo code for deleting domains and types, I don't know how this code will even be hit.  If there isn't a domain or type, the filter will just not do anything, but the return object for delete_domains and delete_sbl_type will always exist. So that logger code wouldn't be hit.  If an exception is hit while trying to delete either, it would be thrown up in the response and the logger code wouldn't be hit.

I'm going to write a story for Nargis to rework the code for the repo or this section to properly check if the domains and types exist and log a warning because yeah I think these loggers are unreachable.

Reply:
Created #40 if you want to check if that properly captures the issue here.